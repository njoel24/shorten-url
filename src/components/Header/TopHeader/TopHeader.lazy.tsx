import React, { lazy, Suspense } from 'react';

const LazyTopHeader = lazy(() => import('./TopHeader'));

const TopHeader = (props: JSX.IntrinsicAttributes & { children?: React.ReactNode; }) => (
  <Suspense fallback={null}>
    <LazyTopHeader {...props} />
  </Suspense>
);

export default TopHeader;
