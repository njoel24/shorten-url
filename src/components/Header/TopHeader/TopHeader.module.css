.TopHeader {
    display: flex;
    width: 90%;
    justify-content: space-between;
    align-items: center;
    color:  var(--gray);
    font-size: 0.9rem;
}

.Left {
    display: flex;
    align-items: center;
}

@media only screen and (max-width: 1024px) {
    .TopHeader {
        justify-content: space-between;
        width: 90%;
    }
}