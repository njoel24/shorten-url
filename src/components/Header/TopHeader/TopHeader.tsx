import React from 'react';
import styles from './TopHeader.module.css';
import {ReactComponent as Logo } from '../../../assets/logo.svg';
import Menu from './Menu/Menu';
import Account from './Account/Account';
import AccountMenuMobile from './AccountMenuMobile/AccountMenuMobile';

const TopHeader = () => (
  <div className={styles.TopHeader} data-testid="TopHeader">
    <div className={styles.Left}>
      <Logo />
      <Menu />
    </div>
    <Account/>
    <AccountMenuMobile />
  </div>
);

export default TopHeader;
