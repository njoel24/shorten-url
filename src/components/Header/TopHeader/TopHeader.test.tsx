import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TopHeader from './TopHeader';

describe('<TopHeader />', () => {
  test('it should mount', () => {
    render(<TopHeader />);
    
    const topHeader = screen.getByTestId('TopHeader');

    expect(topHeader).toBeInTheDocument();
  });

  test('it should match the snapshot', () => {
    const wrapper = render(<TopHeader />);

    expect(wrapper).toMatchSnapshot();
  });
});