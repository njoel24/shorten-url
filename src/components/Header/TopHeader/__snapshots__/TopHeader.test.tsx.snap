// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopHeader /> it should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="TopHeader"
        data-testid="TopHeader"
      >
        <div
          class="Left"
        >
          <svg>
            logo.svg
          </svg>
          <ul
            class="TopHeaderMenu"
            data-testid="Menu"
          >
            <li>
              Features
            </li>
            <li>
              Pricing
            </li>
            <li>
              Resources
            </li>
          </ul>
        </div>
        <ul
          class="Account"
          data-testid="Account"
        >
          <button
            class="Button"
          >
            Login
          </button>
          <button
            class="Button active"
          >
            Sign Up
          </button>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="TopHeader"
      data-testid="TopHeader"
    >
      <div
        class="Left"
      >
        <svg>
          logo.svg
        </svg>
        <ul
          class="TopHeaderMenu"
          data-testid="Menu"
        >
          <li>
            Features
          </li>
          <li>
            Pricing
          </li>
          <li>
            Resources
          </li>
        </ul>
      </div>
      <ul
        class="Account"
        data-testid="Account"
      >
        <button
          class="Button"
        >
          Login
        </button>
        <button
          class="Button active"
        >
          Sign Up
        </button>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<TopHeader /> it should show a logo 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="TopHeader"
        data-testid="TopHeader"
      >
        <div
          class="Left"
        >
          <svg>
            logo.svg
          </svg>
          <ul
            class="TopHeaderMenu"
            data-testid="Menu"
          >
            <li>
              Features
            </li>
            <li>
              Pricing
            </li>
            <li>
              Resources
            </li>
          </ul>
        </div>
        <ul
          class="Account"
          data-testid="Account"
        >
          <button
            class="Button"
          >
            Login
          </button>
          <button
            class="Button active"
          >
            Sign Up
          </button>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="TopHeader"
      data-testid="TopHeader"
    >
      <div
        class="Left"
      >
        <svg>
          logo.svg
        </svg>
        <ul
          class="TopHeaderMenu"
          data-testid="Menu"
        >
          <li>
            Features
          </li>
          <li>
            Pricing
          </li>
          <li>
            Resources
          </li>
        </ul>
      </div>
      <ul
        class="Account"
        data-testid="Account"
      >
        <button
          class="Button"
        >
          Login
        </button>
        <button
          class="Button active"
        >
          Sign Up
        </button>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
